src/lib/autodiscovery/autodiscovery.py

```diff
@@ -140,60 +245,56 @@ async def run_autodiscovery(
     dt_session: ClientSession,
     token: str,
     previously_discovered_metrics: Dict[str, Any],
+    autodiscovery_resources: Dict[str, AutodiscoveryResourceLinking],
+    autodiscovery_metric_block_list: List[str],
 ) -> AutodiscoveryResult:
     start_time = time.time()
     logging_context.log("Adding metrics using autodiscovery")
     metric_context = await get_metric_context(gcp_session, dt_session, token, logging_context)
 
-    filtered_gcp_extension_services = list(
-        filter(lambda x: discovered_resource_type in x.name, gcp_services_list)
+    discovered_metric_descriptors, resource_dimensions = await get_metric_descriptors(
+        metric_context, gcp_session, token, autodiscovery_resources, autodiscovery_metric_block_list
     )
 
-    existing_metric_list = []
-    for service in filtered_gcp_extension_services:
-        existing_metric_list.extend(service.metrics)
+    existing_resources_to_metrics = await get_existing_metrics(autodiscovery_resources)
 
-    discovered_metric_descriptors = await get_metric_descriptors(metric_context, gcp_session, token)
+    autodiscovery_resources_to_metrics = {}
 
-    existing_metric_names = {}
-    for existing_metric in existing_metric_list:
-        existing_metric_names[existing_metric.google_metric] = None
+    for descriptor, project_ids in discovered_metric_descriptors.items():
+        monitored_resource = descriptor.monitored_resources_types[0]
+        if descriptor.value not in existing_resources_to_metrics[monitored_resource]:
+            autodiscovery_resources_to_metrics.setdefault(monitored_resource, []).append(
+                Metric(**(asdict(descriptor)), autodiscovered_metric=True, project_ids=project_ids)
+            )
 
-    missing_metrics_list = []
+    for resource_name, metric_list in autodiscovery_resources_to_metrics.items():
+        logging_context.log(
+            f"For resource: [{resource_name}] discovered {len(metric_list)} metrics"
```

Comment:
What about: Discovered {len(metric_list)} metrics for [{resource_name}] resource.